presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestSpatialPartitioningInternalAggregation.java

```diff
@@ -0,0 +1,163 @@
+        String aggregation = (String) BlockAssertions.getOnlyValue(accumulator.getFinalType(), getFinalBlock(accumulator));
+        assertEquals(aggregation, expectedValue);
+
+        GroupedAccumulator groupedAggregation = accumulatorFactory.createGroupedAccumulator();
+        groupedAggregation.addInput(createGroupByIdBlock(0, page.getPositionCount()), page);
+        String groupValue = (String) getGroupValue(groupedAggregation, 0);
+        assertEquals(groupValue, expectedValue);
+    }
+
+    private InternalAggregationFunction getFunction()
+    {
+        return functionAssertions
+                .getMetadata()
+                .getFunctionRegistry()
+                .getAggregateFunctionImplementation(new Signature("spatial_partitioning",
+                        FunctionKind.AGGREGATE,
+                        TypeSignature.parseTypeSignature(VARCHAR),
+                        TypeSignature.parseTypeSignature(GEOMETRY_TYPE_NAME),
+                        TypeSignature.parseTypeSignature(INTEGER)));
+    }
+
+    private List<OGCGeometry> makeGeometries()
+    {
+        ImmutableList.Builder<OGCGeometry> geometries = ImmutableList.builder();
+        for (int i = 0; i < 10; i++) {
+            for (int j = 0; j < 10; j++) {
+                geometries.add(new OGCPoint(new Point(-10 + i, -10 + j), null));
+            }
+        }
+
+        for (int i = 0; i < 5; i++) {
+            for (int j = 0; j < 5; j++) {
+                geometries.add(new OGCPoint(new Point(-10 + 2 * i, 2 * j), null));
+            }
+        }
+
+        for (int i = 0; i < 4; i++) {
+            for (int j = 0; j < 4; j++) {
+                geometries.add(new OGCPoint(new Point(2.5 * i, -10 + 2.5 * j), null));
+            }
+        }
+
+        for (int i = 0; i < 3; i++) {
+            for (int j = 0; j < 3; j++) {
+                geometries.add(new OGCPoint(new Point(5 * i, 5 * j), null));
+            }
+        }
+
+        return geometries.build();
+    }
+
+    private Block makeGeometryBlock(List<OGCGeometry> geometries)
+    {
+        BlockBuilder builder = GEOMETRY.createBlockBuilder(null, geometries.size());
+        for (OGCGeometry geometry : geometries) {
+            GEOMETRY.writeSlice(builder, serialize(geometry));
+        }
+        return builder.build();
+    }
+
+    private String getSpatialPartitioning(Rectangle extent, List<OGCGeometry> geometries, int partitionCount)
+    {
+        ImmutableList.Builder<Rectangle> rectangles = ImmutableList.builder();
+        for (OGCGeometry geometry : geometries) {
+            Envelope envelope = new Envelope();
+            geometry.getEsriGeometry().queryEnvelope(envelope);
+            rectangles.add(new Rectangle(envelope.getXMin(), envelope.getYMin(), envelope.getXMax(), envelope.getYMax()));
+        }
+
+        return KDB_TREE_CODEC.toJson(buildKDBTree((int) (Math.ceil(geometries.size() * 1.0 / partitionCount)), partitionCount, extent, rectangles.build()));
```

Comment:
`toIntExact`

Reply:
@arhimondr `Math. toIntExact` doesn't take `double`

Reply:
Sorry. I meant to use `toIntExact` instead of `(int)`